cmd/contour/gatewayprovisioner.go

```diff
@@ -41,25 +42,29 @@ func registerGatewayProvisioner(app *kingpin.Application) (*kingpin.CmdClause, *
 	}
 
 	cmd.Flag("contour-image", "The container image used for the managed Contour.").
-		Default(config.contourImage).
-		StringVar(&config.contourImage)
+		Default(provisionerConfig.contourImage).
+		StringVar(&provisionerConfig.contourImage)
 
 	cmd.Flag("envoy-image", "The container image used for the managed Envoy.").
-		Default(config.envoyImage).
-		StringVar(&config.envoyImage)
+		Default(provisionerConfig.envoyImage).
+		StringVar(&provisionerConfig.envoyImage)
 
 	cmd.Flag("metrics-addr", "The address the metric endpoint binds to. It can be set to 0 to disable serving metrics.").
-		Default(config.metricsBindAddress).
-		StringVar(&config.metricsBindAddress)
+		Default(provisionerConfig.metricsBindAddress).
+		StringVar(&provisionerConfig.metricsBindAddress)
+
+	cmd.Flag("leader-election-namespace", "The namespace in which the leader election resource will be created.").
+		Default(config.GetenvOr("CONTOUR_PROVISIONER_NAMESPACE", "projectcontour")).
```

Comment:
to accompany this change, I think we would also like to add an env var configuration like this: https://github.com/projectcontour/contour/blob/491e263e2ab3f4b6e6c0fa629b580491a03ab2da/examples/contour/03-contour.yaml#L77-L82

to [this file](https://github.com/projectcontour/contour/blob/main/examples/gateway-provisioner/03-gateway-provisioner.yaml)

Reply:
done